File: python/869B.py
# 869B. The Eternal Immortality

a, b = map(int, input().split())

def solve(a, b):
    if a == b:
        return 1
    elif a > b:
        return 0
    else:
        ld = 1
        for i in range(a + 1, b + 1):
            ld = (ld * i%10) % 10
            if ld == 0:
                break
    
    return ld

print(solve(a, b))

File: cpp/869B.cpp
// 869B. The Eternal Immortality
// Completed in 31 ms 0 kb

#include<iostream>
 
using namespace std;
 
typedef long long ll;
 
int main() {
    
    ll a, b;
    cin >> a >> b;
    
    if (b - a > 9) {
        cout << 0;
    }
    else {
        ll ld = 1;
        ll i;
        for (i = a + 1; i <= b; i++) {
            ld = (ld * (i % 10)) % 10;
            if (ld == 0) {
                break;
            }
        }
        
        cout << ld << '\n';
    }
    
    return 0;
} 
